server/service/src/sync/api/common_records.rs

```diff
@@ -78,12 +78,17 @@ impl CommonSyncRecord {
             record_data: data,
         } = self;
 
-        let record_id = if action == SyncAction::Merge {
-            // This is (likely) a temporary fix to avoid merge sync records overriding upserts on target table
-            // causing errors as the merge is applied to record that never got upserted first.
-            uuid()
-        } else {
-            record_id
+        let record_id = match action {
+            SyncAction::Merge => {
+                // This is (likely) a temporary fix to avoid merge sync records overriding upserts on target table
+                // causing errors as the merge is applied to record that never got upserted first.
+                uuid()
+            }
+            _ if table_name == *"name_oms_fields" => {
+                // apprehend table name to record_id to avoid name override
```

Comment:
Though should this be `append` lol

Reply:
I snuck it in 😭 
spelling hard

Reply:
@lache-melvin  - ya missed "override" - should be "overwrite"?  ;-)
While I'm here - is table_name a new field in the sync buffer? Is it indexed? If yes, lots of indexed fields in the sync buffer will make it slow to append records to, and I have feels that this is a future performance bottleneck in large systems?

Reply:
@craigdrown, Indexes are on Action, Integration Time, and Integration Error (not sure what the error one is used for...)
record_id is implicitly indexed due being to Primary Key

Reply:
Added to an issue: https://github.com/msupply-foundation/open-msupply/issues/4266